dot/core/messages.go

```diff
@@ -48,7 +50,28 @@ func (s *Service) HandleTransactionMessage(msg *network.TransactionMessage) erro
 		// push to the transaction queue of BABE session
 		hash := s.transactionState.AddToPool(vtx)
 		logger.Trace("Added transaction to queue", "hash", hash)
+
+		// find tx(s) that should not propagate
+		if !val.Propagate {
+			toRemove = append(toRemove, tx)
+		}
```

Comment:
it might be more straightforward to have a list of extrinsics to propagate instead of `toRemove`, and add valid txs with `Propagate == true` to it, that way you don't need to use `findAndDelete`

Reply:
Yes, more straightforward and readable, updated.